src/main/java/codeu/model/store/basic/AdminStore.java

```diff
@@ -0,0 +1,90 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.User;
+import com.google.appengine.repackaged.com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * This class stores admin users, using in-memory data structures to hold values
+ * and automatically loads from and saves to PersistentStorageAgent.
+ * It's a singleton so all servlet classes can access the same instance.
+ *
+ * @author Elle Tojaroon (etojaroo@codeustudents.com)
+ */
+public class AdminStore {
+
+  private static AdminStore instance;
+  private UserStore userStore;
+
+  /** The in-memory admins. */
+  private Map<String, User> admins;
+
+  /** This class is a singleton, so its constructor is private. Call getInstance() instead. */
+  private AdminStore(UserStore userStore) {
+    setUserStore(userStore);
+    admins = new HashMap<>();
```

Comment:
What's the design choice behind having a Hashmap instead of just a list of admins? I don't anticipate there being enough admins for this to have a noticeable benefit but do you suggest we change the way we store our other datatypes to hashmaps?

Reply:
It's quicker for looking up an admin by username as key. You're right that we won't have enough admins to make a difference but I think it's a good practice. :D